File: geoCheat.cpp
#include<bits/stdc++.h>
using namespace std;

typedef long long unsigned ll;
typedef pair<int, int> pll;
vector<pll> A;
int N, a, b;

ll dist(pll &a, pll&b){
    return pow(a.first - b.first, 2) + pow(a.second - b.second, 2);
}

ll solve(){
    ll n = A.size();
    ll  maxDist = 0;
    for (int i = 0; i < n -1; i++){
        for (int j = i + 1; j < n; j++)
            maxDist = max(maxDist, dist(A[i], A[j]));
    }
    return maxDist;
}

int main(){
    ios :: sync_with_stdio(false);
    cin.tie(NULL);
    cin >> N;

    for (int i = 0; i < N; i++){
        cin >> a >> b;
        A.push_back(make_pair(a, b));
        cout << solve() << endl;
    }

    return 0;
}

File: code-py/geoCheat.py

def dist(L):
    def euclid(a, b):
        return pow(a[0] - b[0], 2) + pow(a[1] - b[1], 2)

    if len(L) == 1:
        return 0
    N = len(L)
    maxDist = -1
    for i in xrange(N - 1):
        for j in xrange(i + 1, N):
            maxDist = max(maxDist, euclid(L[i], L[j]))
    return maxDist

if __name__ == '__main__':
    N = input()
    L = []
    for _ in xrange(N):
        L.append(map(int, raw_input().split()))
        print dist(L)
